agent/taskresource/grpcclient/credentialsfetcherclient_test.go

```diff
@@ -111,6 +131,105 @@ func TestCredentialsFetcherClient_AddKerberosLease(t *testing.T) {
 	}
 }
 
+func TestCredentialsFetcherClient_AddNonDomainJoinedKerberosLease(t *testing.T) {
+	tests := []struct {
+		name             string
+		credspecContents []string
+		username         string
+		password         string
+		domain           string
+		response         CredentialsFetcherResponse
+	}{
+		{
+			"invalid request empty credspec contents",
+			[]string{},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			CredentialsFetcherResponse{},
+		},
+		{
+			"invalid request username, password or domain should not be empty",
+			[]string{credspec_webapp01},
+			"",
+			"",
+			"",
+			CredentialsFetcherResponse{},
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			[]string{credspec_webapp01},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			CredentialsFetcherResponse{LeaseID: leaseid, KerberosTicketPaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}},
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Can we please use `require.NoError(t, err)` here instead of `log.Fatal` (and also get rid of the if block)? The latter exits the process.